aws_signing_helper/pkcs11_signer.go

```diff
@@ -0,0 +1,214 @@
+package aws_signing_helper
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"math/big"
+	"reflect"
+
+	"github.com/miekg/pkcs11"
+)
+
+
+type PKCS11Signer struct {
+	//PrivateKey       crypto.PrivateKey
+	cert      *x509.Certificate
+	certChain []*x509.Certificate
+	libpkcs11			 string
+	pinpkcs11			 string
+	idpkcs11			 int
+
+}
+
+func GetMatchingPKCSCerts(certIdentifier CertIdentifier, lib string) (slot int, cert *x509.Certificate, erra error) {
```

Comment:
Minor, but `erra` seems to be a typo here (it isn't used elsewhere).

Reply:
Right. Now using `err error` as done elsewhere.  Removed a differed finalErr that was then being applied by default at the end of function even with no error.